src/software/amazon/ion/impl/lite/IonLoaderLite.java

```diff
@@ -122,23 +122,27 @@ public IonDatagram load(Reader ionText) throws IonException, IOException
 
     public IonDatagram load(byte[] ionData) throws IonException
     {
+        IonReader reader = makeReader(_catalog, ionData, 0, ionData.length, _lstFactory);
         try {
-            IonReader reader = makeReader(_catalog, ionData, 0, ionData.length, _lstFactory);
-            IonDatagramLite datagram = load_helper(reader);
-            return datagram;
+            return load(reader);
         }
-        catch (IOException e) {
-            throw new IonException(e);
+        finally {
+            try {
+                reader.close();
+            }
+            catch (IOException e) {
+                throw new IonException(e);
+            }
         }
+
     }
 
     public IonDatagram load(InputStream ionData)
         throws IonException, IOException
     {
         try {
             IonReader reader = makeReader(_catalog, ionData, _lstFactory);
```

Comment:
please ensure this reader gets closed

Reply:
We can't. If we close the reader we'll close the provided `InputStream` and keeping it open is part of the methods contract. For non Gzipped streams closing the `InputStream` has the same effect as closing the `reader` so it won't cause a leak.